xarray/tests/test_backends.py

```diff
@@ -393,11 +393,29 @@ def test_orthogonal_indexing(self):
                         'dim3': np.arange(5)}
             expected = in_memory.isel(**indexers)
             actual = on_disk.isel(**indexers)
-            self.assertDatasetIdentical(expected, actual)
+            assert_identical(expected, actual)
             # do it twice, to make sure we're switched from orthogonal -> numpy
             # when we cached the values
             actual = on_disk.isel(**indexers)
-            self.assertDatasetIdentical(expected, actual)
+            assert_identical(expected, actual)
+
+    def test_vectorized_indexing(self):
+        # Make sure vectorized_indexing works or at leaset raises
+        # NotImplementedError
+        in_memory = create_test_data()
+        with self.roundtrip(in_memory) as on_disk:
+            indexers = {'dim1': DataArray([0, 2, 0], dims='a'),
+                        'dim2': DataArray([0, 2, 3], dims='a')}
+            expected = in_memory.isel(**indexers)
+            try:
+                actual = on_disk.isel(**indexers)
+                assert_identical(expected, actual)
+                # do it twice, to make sure we're switched from
+                # orthogonal -> numpy when we cached the values
+                actual = on_disk.isel(**indexers)
+                assert_identical(expected, actual)
+            except NotImplementedError:
+                pass
```

Comment:
I think I understand what this test is going for but its not all that clear when its going to pass because the indexing is supported and when its going to pass because it raised a `NotImplementedError`. Can we tease this into two tests somehow?

Reply:
Thanks. I separated the test. Currently, only the dask-related wrappers support vindex.